packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -1154,29 +1158,55 @@ export class ProjectService extends BaseService {
         timezone: string,
         granularity?: DateGranularity,
     ): Promise<CompiledQuery> {
-        const exploreWithOverride = ProjectService.updateExploreWithGranularity(
+        const filteredExplore = ProjectService.filterExplore(
```

Comment:
do the filtering inside the _compilequery method, if there is a FieldReferenceError here, then we check if the file exists in the unfiltered "explore"